### DISCUSSION TITLE
List of index of occurance of words in string but with reverse indexing

### DISCUSSION THREAD

**Discussion by Rinx1er** [TheAlgorithms/Python#8169] [Q&A] [answered]
Hello
I have a long string i want to search for a small string/character and record all indexes in list but with reverse index
Can anyone help me?

**Comment by sourabhkv** [comment, marked as answer]
```python
string="hello how are you i am programming in python , i am also using github"
key="h"#can be single or multi
occurance=[]#this is +ve index

for i in range(len(string)-len(key)+1):
    if string[i:i+len(key)]==key:
        occurance.append(i)

#print(occurance)

occurance_rev = [i-len(string) for i in occurance]
print(occurance_rev)
```

**Reply by sourabhkv** [reply]
you can also use `.rfind` with loop

**Reply by Rinx1er** [reply]
so here u r getting +ve indexes and converting to reverse index

**Reply by sourabhkv** [reply]
yeah

**Comment by trmese** [comment]
> ```python
> string="hello how are you i am programming in python , i am also using github"
> key="h"#can be single or multi
> occurance=[]#this is +ve index
> 
> for i in range(len(string)-len(key)+1):
>     if string[i:i+len(key)]==key:
>         occurance.append(i)
> 
> #print(occurance)
> 
> occurance_rev = [i-len(string) for i in occurance]
> print(occurance_rev)
> ```
